### PR TITLE
[AND-2913] MoPub: Build non-AndroidX version of adapter with 6.7.0 SDK

### PR DESCRIPTION
**Pull request by anton-blashchuk**

### CODE REVIEW

**Comment by HuFangshuai** [comment]
@anton-blashchuk-zoolatech please don't merge this PR to master branch.

File: Vungle/src/main/java/com/mopub/mobileads/AdData.kt [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+import android.os.Parcelable
+import com.mopub.common.Constants
+import com.mopub.common.CreativeOrientation
+import kotlinx.android.parcel.Parcelize
+import java.util.TreeMap
+import kotlin.collections.HashMap
+
+@Parcelize
+data class AdData(
```
Comment (palrahul) [inline Vungle/src/main/java/com/mopub/mobileads/AdData.kt]: @anton-blashchuk-zoolatech where are these Kotlin classes from, are they ported from some version of MoPub's SDK code

**Comment by palrahul** [comment]
@anton-blashchuk-zoolatech suggestion to checkout target branch say 6.7.0-nax-release from 6.7.0-GA tag , and then create this PR from your dev branch into it

File: Vungle/src/main/java/com/mopub/mobileads/AdData.kt (hunk 1 above)
Comment (HuFangshuai) [reply Vungle/src/main/java/com/mopub/mobileads/AdData.kt]: I think this class not related to non-AndroidX patch issue. Can we remove it and Kotlin related gradle configration?

Comment (anton-blashchuk) [reply Vungle/src/main/java/com/mopub/mobileads/AdData.kt]: done

**Comment by anton-blashchuk** [comment]
> @anton-blashchuk-zoolatech please don't merge this PR to master branch.

yes, sure, that was unintended - I think I forgot to update target branch at that point.

File: Vungle/src/main/java/com/mopub/mobileads/VungleAdapterConfiguration.java [hunk 2]
```diff
@@ -42,8 +42,7 @@ public String getAdapterVersion() {
     @Nullable
     @Override
     public String getBiddingToken(@NonNull Context context) {
-        final String bidToken = Vungle.getAvailableBidTokens(10);
-        return bidToken;
+        return null;
```
Comment (HuFangshuai) [inline Vungle/src/main/java/com/mopub/mobileads/VungleAdapterConfiguration.java]: Why removed this bid token logic?

File: Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java [hunk 3]
```diff
@@ -257,60 +266,50 @@ private boolean validateIdsInServerExtras(Map<String, String> serverExtras) {
...
-            if (mPlacementId.equals(placementId)) {
-                MoPubLog.log(getAdNetworkId(), CUSTOM, ADAPTER_NAME, "onAdClick placement id: " + placementId);
                 mHandler.post(new Runnable() {
 
                     @Override
                     public void run() {
-                        if (mInteractionListener != null) {
-                            mInteractionListener.onAdClicked();
+                        if (mCustomEventBannerListener != null) {
+                            mCustomEventBannerListener.onBannerClicked();
```
Comment (alexander-gubov) [inline Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java:296]: we can't call mCustomEventBannerListener.onBannerClicked() in any case, there were params in onAdEnd() to filter but logic was moved to onAdClick()

File: Vungle/src/main/java/com/mopub/mobileads/VungleRewardedVideo.java [hunk 4]
```diff
@@ -237,14 +244,24 @@ private void modifyAdConfig(AdConfig adConfig, VungleMediationConfiguration medi
...
 
-                if (mInteractionListener != null) {
-                    mInteractionListener.onAdDismissed();
-                }
+                MoPubLog.log(getAdNetworkId(), SHOULD_REWARD, ADAPTER_NAME, MoPubReward.NO_REWARD_AMOUNT,
+                        MoPubReward.NO_REWARD_LABEL);
+                // Vungle does not provide a callback when a user should be rewarded.
+                // You will need to provide your own reward logic if you receive a reward with
+                // "NO_REWARD_LABEL" && "NO_REWARD_AMOUNT"
+                MoPubRewardedVideoManager.onRewardedVideoCompleted(VungleRewardedVideo.class,
```
Comment (palrahul) [inline Vungle/src/main/java/com/mopub/mobileads/VungleRewardedVideo.java:258]: shouldnt this be in onAdRewarded

Comment (palrahul) [inline Vungle/src/main/java/com/mopub/mobileads/VungleRewardedVideo.java]: as there is separate onRewardedVideoClosed. , should the onRewardedVideoCompleted be called from onAdRewarded.

File: Vungle/src/main/java/com/mopub/mobileads/VungleRewardedVideo.java [hunk 5]
```diff
@@ -253,27 +270,17 @@ public void onAdEnd(String placementId) {
...
-                        MoPubReward.NO_REWARD_LABEL);
-                // Vungle does not provide a callback when a user should be rewarded.
-                // You will need to provide your own reward logic if you receive a reward with
-                // "NO_REWARD_LABEL" && "NO_REWARD_AMOUNT"
-                if (mInteractionListener != null) {
-                    mInteractionListener.onAdComplete(MoPubReward.success(MoPubReward.NO_REWARD_LABEL,
-                            MoPubReward.NO_REWARD_AMOUNT));
-                }
-            }
+            MoPubRewardedVideoManager.onRewardedVideoClicked(VungleRewardedVideo.class,
```
Comment (palrahul) [inline Vungle/src/main/java/com/mopub/mobileads/VungleRewardedVideo.java]: not sure if this belongs here

File: Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java [hunk 6]
```diff
@@ -23,21 +21,22 @@
 import java.util.Map;
 import java.util.concurrent.atomic.AtomicBoolean;
 
-import static com.mopub.common.DataKeys.ADUNIT_FORMAT;
+import static com.mopub.network.ImpressionData.ADUNIT_FORMAT;
```
Comment (HuFangshuai) [inline Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java:24]: Pls use `import static com.mopub.common.DataKeys.ADUNIT_FORMAT;`

File: Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java [hunk 7]
```diff
@@ -121,8 +130,8 @@ public void run() {
         mAdConfig.setAdSize(vungleAdSize);
         sVungleRouter.addRouterListener(mPlacementId, mVungleRouterListener);
 
-        VungleMediationConfiguration.adConfigWithExtras(mAdConfig, extras);
-        if (VungleMediationConfiguration.isStartMutedNotConfigured(extras)) {
+        VungleMediationConfiguration.adConfigWithLocalExtras(mAdConfig, localExtras);
+        if (VungleMediationConfiguration.isStartMutedNotConfiguredLocal(localExtras)) {
```
Comment (HuFangshuai) [inline Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java]: There's no difference between `isStartMutedNotConfiguredLocal (localExtras)` and `isStartMutedNotConfigured(localExtras)`. Suggestion to reuse it.
```
if (VungleMediationConfiguration.isStartMutedNotConfigured(localExtras)) {
```

File: Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java [hunk 8]
```diff
@@ -148,30 +157,36 @@ public void run() {
...
+        String adUnitFormat = serverExtras.get(ADUNIT_FORMAT);
         if (!TextUtils.isEmpty(adUnitFormat)) {
             adUnitFormat = adUnitFormat.toLowerCase();
         }
-        final boolean isMRECFormat = "medium_rectangle".equals(adUnitFormat);
-        if (isMRECFormat) {
-            if (adWidthInDp >= VUNGLE_MREC.getWidth() && adHeightInDp >= VUNGLE_MREC.getHeight()) {
-                adSizeType = VUNGLE_MREC;
-            }
+        if (adWidthInDp >= VUNGLE_MREC.getWidth() && adHeightInDp >= VUNGLE_MREC.getHeight()) {
```
Comment (HuFangshuai) [inline Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java:188]: Pls keep `isMRECFormat` logic here, `adWidth` and `adHeight` match VUNGLE_MREC not means this ad unit is MREC ad format.
```
final boolean isMRECFormat = "medium_rectangle".equals(adUnitFormat);
        if (isMRECFormat) {
            if (adWidthInDp >= VUNGLE_MREC.getWidth() && adHeightInDp >= VUNGLE_MREC.getHeight()) {
                adSizeType = VUNGLE_MREC;
            }
        } else {
        ...
        }
```

File: Vungle/src/main/java/com/mopub/mobileads/VungleMediationConfiguration.java [hunk 9]
```diff
@@ -120,11 +120,53 @@ static void adConfigWithExtras(@NonNull final AdConfig adConfig,
         }
     }
 
+    static void adConfigWithLocalExtras(AdConfig adConfig, Map<String, Object> localExtras) {
```
Comment (HuFangshuai) [inline Vungle/src/main/java/com/mopub/mobileads/VungleMediationConfiguration.java:77]: Seems this method same as `adConfigWithLocalExtras()`, can we reuse it?

File: Vungle/src/main/java/com/mopub/mobileads/VungleMediationConfiguration.java [hunk 10]
```diff
@@ -120,11 +120,53 @@ static void adConfigWithExtras(@NonNull final AdConfig adConfig,
...
+        }
+    }
+
+
     static boolean isStartMutedNotConfigured(@NonNull final Map<String, String> extras) {
         return !extras.containsKey(Builder.EXTRA_START_MUTED_KEY) &&
                 !extras.containsKey(Builder.EXTRA_SOUND_ENABLED_KEY);
     }
 
+    static boolean isStartMutedNotConfiguredLocal(Map<String, Object> localExtras) {
```
Comment (HuFangshuai) [inline Vungle/src/main/java/com/mopub/mobileads/VungleMediationConfiguration.java]: Same as `isStartMutedNotConfigured ()`, suggestion to reuse it.

File: Vungle/src/main/java/com/mopub/mobileads/VungleRewardedVideo.java (hunk 4 above)
Comment (HuFangshuai) [reply Vungle/src/main/java/com/mopub/mobileads/VungleRewardedVideo.java]: We can remove code parts from line 253 to line 261.

File: Vungle/src/main/java/com/mopub/mobileads/VungleRewardedVideo.java (hunk 5 above)
Comment (HuFangshuai) [reply Vungle/src/main/java/com/mopub/mobileads/VungleRewardedVideo.java]: Pls replace this event implementaion with following code, we should use `onRewardedVideoCompleted() `:
```
if (mPlacementId.equals(placementId)) {
	MoPubLog.log(getAdNetworkId(), SHOULD_REWARD, ADAPTER_NAME, MoPubReward.NO_REWARD_AMOUNT,
	        MoPubReward.NO_REWARD_LABEL);
	// Vungle does not provide a callback when a user should be rewarded.
	// You will need to provide your own reward logic if you receive a reward with
	// "NO_REWARD_LABEL" && "NO_REWARD_AMOUNT"
	MoPubRewardedVideoManager.onRewardedVideoCompleted(VungleRewardedVideo.class,
	        mPlacementId,
	        MoPubReward.success(MoPubReward.NO_REWARD_LABEL,
	                MoPubReward.NO_REWARD_AMOUNT));
}
```

File: Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java (hunk 8 above)
Comment (alexander-gubov) [reply Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java:188]: @HuFangshuai The issue is that "medium_rectangle" is not passed in old Mopub as far as I remember.

Comment (HuFangshuai) [reply Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java:188]: Thank @alexander-gubov, yes, MoPub 5.8.0 SDK does not pass `medium_rectangle ` to adapter.

File: Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java (hunk 6 above)
Comment (anton-blashchuk) [reply Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java:24]: it's not there to use.

Comment (HuFangshuai) [reply Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java:24]: yes, MoPub 5.8.0 SDK not contains `DataKeys.ADUNIT_FORMAT`.

File: Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java (hunk 7 above)
Comment (anton-blashchuk) [reply Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java]: they have different parameter type. I'll change type for isStartMutedNotConfigured's parameter to Map<String, Object>, and remove isStartMutedNotConfiguredLocal

File: Vungle/src/main/java/com/mopub/mobileads/VungleMediationConfiguration.java (hunk 9 above)
Comment (anton-blashchuk) [reply Vungle/src/main/java/com/mopub/mobileads/VungleMediationConfiguration.java:77]: almost same except parameter. I'll just remove adConfigWithExtras which needs values in Map to be Strings

File: Vungle/src/main/java/com/mopub/mobileads/VungleMediationConfiguration.java (hunk 10 above)
Comment (anton-blashchuk) [reply Vungle/src/main/java/com/mopub/mobileads/VungleMediationConfiguration.java]: again different non-exchangeable(one way) parameter type.

**Review by HuFangshuai — APPROVED** [review]
LGTM

**Review by palrahul — APPROVED** [review]
lgtm

**Review by jiancchen — APPROVED** [review]

**Pull request merged**